packages/fleather/lib/src/widgets/controller.dart

```diff
@@ -32,6 +46,12 @@ class FleatherController extends ChangeNotifier {
   ParchmentStyle get toggledStyles => _toggledStyles;
   ParchmentStyle _toggledStyles = ParchmentStyle();
 
+  /// Returns true if there is at least one undo operation.
+  bool get canUndo => _history.canUndo;
+
+  /// Returns true if there is at least one redo operation.
+  bool get canRedo => _history.canRedo;
+
```

Comment:
I think it's better to move these to `HistoryHandler` extension too. WDYT? @amantoux